### PR TITLE
Added more audit logs

### PR DESCRIPTION
**Pull request by EldoreiJK**
This should provide necessary logs to get some idea about the auditing results.
This is dependant on https://github.com/42ity/fty-common-rest/pull/36 and won't pass until it is merged and updated on Jenkins/Travis

### CODE REVIEW

File: src/web/src/alert_ack.ecpp [hunk 1]
```diff
@@ -59,14 +59,19 @@ bool database_ready;
     if (!database_ready) {
         log_debug ("Database is not ready yet.");
         std::string err = TRANSLATE_ME ("Database is not ready yet, please try again after a while.");
+        if (request.getMethod () == "PUT")
+            log_info_audit ("Request PUT alert ack FAILED");
```
Comment (jeffdegott) [inline src/web/src/alert_ack.ecpp]: Just a comment. On failure treatment, we invoke log_info_audit(). It should be log_error_audit().

**Comment by jimklimov** [comment]
Missed some headers?

````
  CXX      src/shared/src_libfty_rest_la-magic.lo
ecppc --nolog -I./include -I./include -o ./src/web/src/add_gpio.cc ./src/web/src/add_gpio.ecpp && \
	    mv -f ./src/web/src/add_gpio.cc.cpp ./src/web/src/add_gpio.cc
  CXX      src/web/src/src_libfty_rest_la-add_gpio.lo
./src/web/src/add_gpio.ecpp: In member function 'virtual unsigned int {anonymous}::_component_::operator()(tnt::HttpRequest&, tnt::HttpReply&, tnt::QueryParams&)':
./src/web/src/add_gpio.ecpp:154:104: error: 'CHECK_USER_PERMISSIONS_OR_DIE_AUDIT' was not declared in this scope
     CHECK_USER_PERMISSIONS_OR_DIE_AUDIT (PERMISSIONS, audit_msg.empty () ? nullptr : audit_msg.c_str ());
                                                                                                        ^
make[1]: *** [src/web/src/src_libfty_rest_la-add_gpio.lo] Error 1
make[1]: Leaving directory `/home/travis/build/42ity/fty-rest'
````

Otherwise LGTM ;)

File: src/web/src/alert_ack.ecpp (hunk 1 above)
Comment (EldoreiJK) [reply src/web/src/alert_ack.ecpp]: Updated, thx

File: src/web/src/auth.ecpp [hunk 2]
```diff
@@ -53,8 +53,8 @@ std::string checked_token;
     std::string token = qparam.param ("token");
     std::string audit_msg;
     if (request.getMethod () == "POST")
-        audit_msg = std::string ("Request CREATE auth username ") + username + " FAILED";
-    check_regex_text_or_die ("token", token, checked_token, "^[-/._~+/a-z0-9]{0,255}={0,10}$",
+        audit_msg = std::string ("Request CREATE auth FAILED");
+    check_regex_text_or_die_audit ("token", token, checked_token, "^[-/._~+/a-z0-9]{0,255}={0,10}$",
```
Comment (jimklimov) [inline src/web/src/auth.ecpp]: For future: macro names should be all same-style, commonly upper-cased.

Also, the `CHECK_USER_PERMISSIONS_OR_DIE_AUDIT` above casts `(char *) nullptr` while other cases I've seen recently do not...

**Pull request merged**